wrappers/clang-target-wrapper.c

```diff
@@ -192,8 +192,9 @@ int _tmain(int argc, TCHAR* argv[]) {
         exec_argv[arg++] = _T("-DUNICODE");
         // add the minimum runtime to use for UWP targets
         exec_argv[arg++] = _T("-Wl,-lwindowsapp");
-        // This requires that the default crt is ucrt.
-        exec_argv[arg++] = _T("-Wl,-lvcruntime140_app");
+        // This still requires that the default crt is ucrt (so libc++.a has
+        // been built for that CRT).
+        exec_argv[arg++] = _T("-Wl,-lucrtapp");
```

Comment:
Given ucrtapp contains ucrt (minus the -private stuff) is this needed to link both ? Or you mean the toolchain crt should be ucrt ?
Other than the comment, the set LGTM.
You may also mension it works because in Clang the CRT in use is detected with :
```cpp
  for (auto Lib : Args.getAllArgValues(options::OPT_l))
    if (StringRef(Lib).startswith("msvcr") || StringRef(Lib).startswith("ucrt"))
      return;
  CmdArgs.push_back("-lmsvcrt");
```
So even if it's not the pure original name, it still works.

Reply:
No, it's not needed to link both - and it won't (exactly due to the code in clang that you quoted).

I just mean that even if we force UCRT here, both when linking and when compiling, we still need the toolchain to have been built with UCRT as default (no difference if the default was ucrtbase, ucrt or ucrtapp), so that libc++.a targets the right kind of CRT.

Reply:
Would it be clearer if I reformulated it like this? "This still requires that the toolchain (in particular, libc++.a) has been built targeting UCRT originally."

Reply:
Yes